### ISSUE TITLE
Pod scheduling failing with empty reason

### ISSUE THREAD

**Issue by antoineco** [kubernetes/kubernetes#17488]
Kubernetes v1.1.1

I'm facing a situation where all my pods (~80) stay in a *Pending* state for no apparent reason, **except** DaemonSet pods:

```
default            mailhog-ycokx       0/1       Pending   0          1h        
ft-105             api-oppa2           0/1       Pending   0          1h        
ft-105             cache-2cv3e         0/1       Pending   0          37m       
ft-105             db-rjuo2            0/1       Pending   0          1h        
ft-105             sidekiq-3psuo       0/1       Pending   0          1h        
ft-105             web-mfpov           0/1       Pending   0          1h        
kube-system        kube-dns-v9-m0k43   0/4       Pending   0          1h        
kube-system        kube-ui-v2-m1y3f    0/1       Pending   0          1h        
kube-system        nrsysmond-0gf77     1/1       Running   0          23m       ip-10-0-12-147.eu-west-1.compute.internal
kube-system        nrsysmond-da0ix     1/1       Running   0          23m       ip-10-0-12-111.eu-west-1.compute.internal
kube-system        nrsysmond-lsgtm     1/1       Running   0          23m       ip-10-0-12-110.eu-west-1.compute.internal
...
```

I have 3 *Ready* nodes but `kube-scheduler` keeps throwing errors like:

```
kube-scheduler[11670]: I1119 08:19:13.722615   11670 factory.go:220] About to try and schedule pod web-mz1d4
kube-scheduler[11670]: I1119 08:19:13.722878   11670 scheduler.go:127] Failed to schedule: &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:web-mz1d4 GenerateName:web- Namespace:wear-banners SelfLink:/api/v1/namespaces/wear-banners/pods/web-mz1d4 UID:54948e88-8e94-11e5-8f22-0a4c87eff515 ResourceVersion:21331526 Generation:0 CreationTimestamp:2015-11-19 08:05:42 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[tier:web app:web] Annotations:map[kubernetes.io/created-by:{"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"wear-banners","name":"web","uid":"90a40100-8a0a-11e5-8908-0a4c87eff515","apiVersion":"v1","resourceVersion":"18372610"}}
kube-scheduler[11670]: ]} Spec:{Volumes:[{Name:default-token-4nf1q VolumeSource:{HostPath:<nil> EmptyDir:<nil> GCEPersistentDisk:<nil> AWSElasticBlockStore:<nil> GitRepo:<nil> Secret:0xc208716230 NFS:<nil> ISCSI:<nil> Glusterfs:<nil> PersistentVolumeClaim:<nil> RBD:<nil> Cinder:<nil> CephFS:<nil> Flocker:<nil> DownwardAPI:<nil> FC:<nil>}}] Containers:[{Name:web Image:docker-dev.freeletics.com/web:FT-118-Wear-Banners Command:[] Args:[] WorkingDir: Ports:[{Name:http HostPort:0 ContainerPort:8080 Protocol:TCP HostIP:}] Env:[{Name:ENV Value:development ValueFrom:<nil>} {Name:API_SERVER Value:http://qa.freeletics.com:31146 ValueFrom:<nil>}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:default-token-4nf1q ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:<nil> ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log ImagePullPolicy:Always SecurityContext:<nil> Stdin:false StdinOnce:false TTY:false}] RestartPolicy:Always TerminationGracePeriodSeconds:0xc208716218 ActiveDeadlineSeconds:<nil> DNSPolicy:ClusterFirst NodeSelector:map[] ServiceAccountName:default NodeName: HostNetwork:false HostPID:false HostIPC:false ImagePullSecrets:[{Name:docker-registry}]} Status:{Phase:Pending Conditions:[] Message: Reason: HostIP: PodIP: StartTime:<nil> ContainerStatuses:[]}}
kube-scheduler[11670]: I1119 08:19:13.722976   11670 generic_scheduler.go:47] failed to find fit for pod web-mz1d4 on node ip-10-0-12-110.eu-west-1.compute.internal: PodFitsResources
kube-scheduler[11670]: I1119 08:19:13.723009   11670 generic_scheduler.go:47] failed to find fit for pod web-mz1d4 on node ip-10-0-12-110.eu-west-1.compute.internal:
kube-scheduler[11670]: E1119 08:19:13.723005   11670 factory.go:273] Error scheduling wear-banners web-mz1d4: Failed for reason  and possibly others; retrying
```

All 3 nodes have the same status:

```
  apiVersion: v1
  kind: Node
  metadata:
    creationTimestamp: 2015-10-11T19:42:16Z
    labels:
      kubernetes.io/hostname: ip-10-0-12-110.eu-west-1.compute.internal
    name: ip-10-0-12-110.eu-west-1.compute.internal
    resourceVersion: "21360737"
    selfLink: /api/v1/nodes/ip-10-0-12-110.eu-west-1.compute.internal
    uid: 2d2c53da-7050-11e5-933d-0a4c87eff515
  spec:
    externalID: i-xxx
    providerID: aws:///eu-west-1a/i-xxx
  status:
    addresses:
    - address: 10.0.12.110
      type: InternalIP
    - address: 10.0.12.110
      type: LegacyHostIP
    capacity:
      cpu: "2"
      memory: 8179028Ki
      pods: "40"
    conditions:
    - lastHeartbeatTime: 2015-11-19T09:18:01Z
      lastTransitionTime: 2015-11-19T09:08:17Z
      message: kubelet is posting ready status
      reason: KubeletReady
      status: "True"
      type: Ready
    - lastHeartbeatTime: 2015-11-19T09:18:01Z
      lastTransitionTime: 2015-11-02T17:11:08Z
      message: kubelet has sufficient disk space available
      reason: KubeletHasSufficientDisk
      status: "False"
      type: OutOfDisk
    daemonEndpoints:
      kubeletEndpoint:
        Port: 10250
    nodeInfo:
      bootID: 12fb6a2c-f1b7-4213-bc40-774d23d46525
      containerRuntimeVersion: docker://1.8.3
      kernelVersion: 4.2.2-coreos-r1
      kubeProxyVersion: v1.1.1
      kubeletVersion: v1.1.1
      machineID: c7d61cdb151245c29113f873bde4b3e8
      osImage: CoreOS 845.0.0
      systemUUID: EC2C41E0-9038-0549-624D-3423E6C1A374
```

I've tried to restart components one by one, also to reboot the system, but nothings seems to work, the cluster is just frozen atm.

**Comment by davidopp** [comment]
Hi @antoineco . Sorry for the delay in replying.

The DaemonSet pods don't get scheduled by the regular Kubernetes scheduler (they are scheduled directly by the DaemonSet controller), which explains why the DaemonSet pods are getting scheduled but the other ones aren't.

But something seems to be going on with the scheduler. Is your problem reproducible? If so, can you run it again and provide the output of `kubectl describe node` from each of the nodes? Also, am I correct in understanding that the pods you are scheduling request no resources (I see `resources:{Limits:map[] Requests:map[]}` in the output you pasted).

(It would be even better if you can reproduce it with a smaller number of pods.)

Did you change the predicate functions that are running, or you are just using the default ones?

**Comment by antoineco** [comment]
Hi @davidopp, I'm not sure if I can reproduce the issue but I can try to remember the state of the cluster when I reported this. 

In the morning I found my api-server unresponsive as many TLS handshake errors were polluting the logs, making the load average very high on the instance (lots of I/O wait, this master has a very slow spinning disk). Stopping everything and restarting the services one after the other helped me bringing my cluster back on track, but then the I/O wait went high again while the scheduler was throwing the aforementioned errors. Eventually I managed to get my pods scheduled after deleting half of the replication controllers.

Some of my pods have resource limits but most don't. 

This cluster is scaled down on purpose (qa purpose) but behaved nevertheless pretty well until this incident. Having a bottleneck on the hard drive may have contributed to the situation, I'll try to reproduce in similar conditions and will report back here.

**Comment by antoineco** [comment]
@davidopp it happened again last night. I found 2 of my 3 nodes in a "Not Ready" state with both `kube-proxy` and `kubelet` services down and all pods in a Pending or Terminating state. Nodes didn't restart, apparently only the k8s node services stopped and didn't restart, despite the watchdog set on the systemd unit (*Restart=always*). I'm surprised that no pod kept running despite the fact that one node was still Ready. Currently all 3 nodes are Ready and the load average is extremely high on my master due to high network traffic.

I'm trying to dig into the logs right now, I'll post my updates here.

**Comment by antoineco** [comment]
I got it. It appears to be related to the hardware. At 10PM the t2 instance (AWS) on which my k8s master components are running had consumed all its CPU credits, which led to the high load average I observed:

![cpu_credit](https://cloud.githubusercontent.com/assets/3299086/11587024/050e5e18-9a77-11e5-82c5-48dd8086edde.png)
![load_avg](https://cloud.githubusercontent.com/assets/3299086/11587109/824e02f2-9a77-11e5-97d6-2f3c224f18f3.png)

As a result kubelet/kube-proxy logs are full of timeouts, 500 errors and so on. Not sure if you want me to go more into details.

Anyway, I think I'll send a PR to add a warning about using t2 instances on AWS.
